### PR TITLE
fix: Ivy Failing Test: paddle - searching.where

### LINKED ISSUE #28409: Fix Ivy Failing Test: paddle - searching.where
To-Do list #27501

### PR DESCRIPTION
**Pull request by MuhammadNizamani**
# PR Description
passing locally on 200 examples 
paddle-where

## Related Issue

Closes #28409

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

File: ivy/functional/backends/paddle/searching.py [hunk 1]
```diff
@@ -119,6 +119,9 @@ def nonzero(
     return res.T
 
 
+@with_supported_dtypes(
+    {"2.6.0 and below": ("in32", "int64", "float32", "float64")}, backend_version
```
Comment (Ishticode) [inline ivy/functional/backends/paddle/searching.py:123]: did you mean for the first to be `int32` not `in32`.

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @MuhammadNizamani 
Thank you for the PR :)
Note that with supported dtypes decorator means that these are the only suppported types. The functions seems to be trying to handle several other dtypes under
```python
    if ret_dtype in [
        paddle.int8,
        paddle.int16,
        paddle.uint8,
        paddle.float16,
        paddle.bool,
    ]:
```
and also under
```python
elif ret_dtype in [paddle.complex64, paddle.complex128]:
```
The decorator seems to have negated all these but the function is still trying to handle. Could we fix this? Let me know if something is unclear. Thanks :)

**Comment by MuhammadNizamani** [comment]
@Ishticode  sir first of all I love your work in IVY. and here is my argument ,  when I run paddle where function without decorator it says paddle where function support  ` float32, float64, int32, int64.` and I also checked paddlepaddle docs for it says the same thing.
docs link for paddlepaddle function
 https://www.paddlepaddle.org.cn/documentation/docs/en/2.2/api/paddle/where_en.html#where
 And this is SS for it .
paddle where datatypme

Thanks for the review

File: ivy/functional/backends/paddle/searching.py (hunk 1 above)
Comment (MuhammadNizamani) [reply ivy/functional/backends/paddle/searching.py:123]: It is a typo I am  going to fix it .

**Comment by MuhammadNizamani** [comment]
@Ishticode  I have gone through again and I checked  that `float16` and `bfloat16` keep failing  so I just make them unsupported.
if anything I am missing please enlighten me.
Thanks for your informative response.

**Review by Ishticode — APPROVED** [review]
Thanks @MuhammadNizamani for your continous work :)

**Pull request merged**